lib/devise_security_extension/hooks/session_limitable.rb

```diff
@@ -3,7 +3,7 @@
 # and on authentication. Retrieving the user from session (:fetch) does
 # not trigger it.
 Warden::Manager.after_set_user :except => :fetch do |record, warden, options|
-  if record.respond_to?(:update_unique_session_id!) && warden.authenticated?(options[:scope])
+  if record.respond_to?(:update_unique_session_id!) && warden.authenticated?(options[:scope]) && !warden.request.env['devise.skip_session_limitable']  && !warden.request.session.key?('impersonator_id')
```

Comment:
That's quite a conditional. Can you refactor into smaller, meaningfully-named methods?